account_invoice_split/wizard/account_invoice_split.py

```diff
@@ -0,0 +1,164 @@
+    ('name', 'name'),
+    ('quantity', 'total_quantity')]
+
+
+class AccountInvoiceSplit(models.TransientModel):
+    _name = 'account.invoice.split'
+
+    line_ids = fields.One2many(
+        comodel_name='account.invoice.split.line', inverse_name='wizard_id',
+        string='Invoice lines to Split')
+
+    @api.model
+    def _get_invoice_id_from_context(self):
+        active_ids = self.env.context.get('active_ids')
+        if len(active_ids) != 1:
+            raise exceptions.UserError(
+                _("You can call this action only for a single invoice."))
+        return active_ids[0]
+
+    @api.model
+    def _get_invoice_from_context(self):
+        invoice_id = self._get_invoice_id_from_context()
+        invoice = self.env['account.invoice'].browse(invoice_id)
+        self._check_invoice(invoice)
+        return invoice
+
+    @api.model
+    def _check_invoice(self, invoice):
+        if invoice.state != 'draft':
+            raise UserError(
+                _('Invoice must be draft, not %s') % (invoice.state,))
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super(AccountInvoiceSplit, self).default_get(fields_list)
+        invoice = self._get_invoice_from_context()
+        split_lines = []
+        for line in invoice.invoice_line_ids:
+            current_line_values = {
+                'origin_invoice_line_id': line.id,
+                'product_id': line.product_id.id,
+                'name': line.name,
+                'total_quantity': line.quantity,
+            }
+            split_lines.append((0, 0, current_line_values))
+        res['line_ids'] = split_lines
+        return res
+
+    @api.model
+    def _get_invoice_values(self, invoice):
+        """Return default value for copy method.
+        Can be override to add some fields"""
+        return {'invoice_line_ids': []}
+
+    @api.model
+    def _create_invoice(self, invoice_to_split, invoice_lines):
+        new_invoice = False
+        if invoice_lines:
+            default = self._get_invoice_values(invoice_to_split)
+            default['invoice_line_ids'] = invoice_lines
+            new_invoice = invoice_to_split.copy(default=default)
+        if not new_invoice:
+            raise exceptions.Warning(
+                _("There is nothing to split. Please fill "
+                  "the 'quantities to split' column."))
+        return new_invoice
+
+    @api.multi
+    def _split_invoice(self):
+        self.ensure_one()
+
+        invoice_to_split = self._get_invoice_from_context()
+        invoice_lines = []
+        for line in self.line_ids:
+            invoice_line = line.origin_invoice_line_id
+            if not float_is_zero(line.quantity_to_split):
+                # I Check if the quantity to split isn't greater then the
+                # quantity on the origin line
+                qty_is_valid = float_compare(
+                    line.quantity_to_split, invoice_line.quantity) == -1
```

Comment:
@benwillig It produces a stacktrace because of the missing precision (`precision_digits`). Use `self.env['decimal.precision'].precision_get('Product Unit of Measure')`

Reply:
Thanks, I fixed it.